enhancements/authentication/allowing-uri-scheme-in-oidc-sub-claims.md

```diff
@@ -0,0 +1,167 @@
+---
+title: allowing-uri-scheme-in-oidc-sub-claims
+authors:
+  - "@stlaz"
+reviewers:
+  - "@stts"
+  - "@deads2k"
+approvers:
+  - "@stts"
+  - "@deads2k"
+creation-date: 2020-01-19
+last-updated: 2020-01-19
+status: implementable
+see-also:
+  -
+replaces:
+  -
+superseded-by:
+  -
+---
+
+# Allowing URI Scheme in OIDC `sub` claims
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Make the oauth-server capable of setting up identities for OIDC identity
+providers that use the URI scheme in the `sub` claim of their ID tokens.
+
+## Motivation
+
+Per the [OIDC specification](https://openid.net/specs/openid-connect-core-1_0-final.html#IDToken)
+of the ID token and its more thorough details described in
+[RFC7519 - JSON Web Token (JWT)](https://tools.ietf.org/html/rfc7519#section-4.1.2),
+the `sub` claim may be represented as both a string or a URI. However, URI by
```

Comment:
nit: an URI is a string.

Reply:
the linked RFC separates them in this sense:
```
   StringOrURI
      A JSON string value, with the additional requirement that while
      arbitrary string values MAY be used, any value containing a ":"
      character MUST be a URI [RFC3986].  StringOrURI values are
      compared as case-sensitive strings with no transformations or
      canonicalizations applied.

```

Reply:
please clarify in the text here